R/aes-evaluation.r

```diff
@@ -93,31 +93,42 @@ is_dotted_var <- function(x) {
 }
 
 # Determine if aesthetic is calculated
-is_calculated_aes <- function(aesthetics) {
-  vapply(aesthetics, is_calculated, logical(1), USE.NAMES = FALSE)
+is_calculated_aes <- function(aesthetics, warn = FALSE) {
+  vapply(aesthetics, is_calculated, warn = warn, logical(1), USE.NAMES = FALSE)
 }
 is_scaled_aes <- function(aesthetics) {
   vapply(aesthetics, is_scaled, logical(1), USE.NAMES = FALSE)
 }
 is_staged_aes <- function(aesthetics) {
   vapply(aesthetics, is_staged, logical(1), USE.NAMES = FALSE)
 }
-is_calculated <- function(x) {
+is_calculated <- function(x, warn = FALSE) {
   if (is_call(get_expr(x), "after_stat")) {
     return(TRUE)
   }
   # Support of old recursive behaviour
   if (is.atomic(x)) {
     FALSE
   } else if (is.symbol(x)) {
-    is_dotted_var(as.character(x))
+    res <- is_dotted_var(as.character(x))
+    if (res && warn) {
+      # TODO: if we want to set always = TRUE, we probably need some mechanism
```

Comment:
I think you can remove this comment; we're still some time away from setting `always = TRUE`, if we ever do so. This is a very old feature, so I think the deprecation needs to be quite mild.

Reply:
Sure.